libraries/api-extractor-model/src/mixins/ApiExtendsMixin.ts

```diff
@@ -0,0 +1,393 @@
+  extendsTokenRanges: IExcerptTokenRange[];
+}
+
+const _extendsTypes: unique symbol = Symbol('ApiExtendsMixin._extendsTypes');
+
+/**
+ * The mixin base class for API items that can extend other API items.
+ *
+ * @remarks
+ *
+ * This is part of the {@link ApiModel} hierarchy of classes, which are serializable representations of
+ * API declarations.  The non-abstract classes (e.g. `ApiClass`, `ApiEnum`, `ApiInterface`, etc.) use
+ * TypeScript "mixin" functions (e.g. `ApiDeclaredItem`, `ApiItemContainerMixin`, etc.) to add various
+ * features that cannot be represented as a normal inheritance chain (since TypeScript does not allow a child class
+ * to extend more than one base class).  The "mixin" is a TypeScript merged declaration with three components:
+ * the function that generates a subclass, an interface that describes the members of the subclass, and
+ * a namespace containing static members of the class.
+ *
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export interface ApiExtendsMixin extends ApiItem {
+  /**
+   * The types referenced via an "extends" heritage clause.
+   */
+  readonly extendsTypes: HeritageType[];
+
+  /**
+   * Finds all of the ApiItem's immediate and inherited members by walking up the inheritance tree.
+   */
+  findMembersWithInheritance(
+    messageCallback?: FindMembersWithInheritanceMessageCallback
+  ): IFindMembersWithInheritanceResult;
+
+  /** @override */
+  serializeInto(jsonObject: Partial<IApiExtendsMixinJson>): void;
+}
+
+/**
+ * Mixin function for {@link (ApiExtendsMixin:interface)}.
+ *
+ * @param baseClass - The base class to be extended
+ * @returns A child class that extends baseClass, adding the {@link (ApiExtendsMixin:interface)} functionality.
+ *
+ * @public
+ */
+export function ApiExtendsMixin<TBaseClass extends IApiItemConstructor>(
+  baseClass: TBaseClass
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): TBaseClass & (new (...args: any[]) => ApiExtendsMixin) {
+  class MixedClass extends baseClass implements ApiExtendsMixin {
+    public readonly [_extendsTypes]: HeritageType[] = [];
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public constructor(...args: any[]) {
+      super(...args);
+
+      const options: IApiExtendsMixinOptions = args[0];
+
+      if (this instanceof ApiDeclaredItem) {
+        for (const extendsTokenRange of options.extendsTokenRanges) {
+          this.extendsTypes.push(new HeritageType(this.buildExcerpt(extendsTokenRange)));
+        }
+      } else {
+        throw new InternalError('ApiExtendsMixin expects a base class that inherits from ApiDeclaredItem');
+      }
+    }
+
+    /**
+     * Finds all of the ApiItem's immediate and inherited members by walking up the inheritance tree.
+     *
+     * @param messageCallback - A callback that is called when this method hits some kind of error
+     * that could impact finding inherited members.
```

Comment:
```suggestion
     * @param messageCallback - A callback that reports diagnostic information about the
     * search operation.
```

I'd like to keep open the possibility that `messageCallback()` might report informational messages that do not indicate any error.  For example, if the crawl becomes more complicated in the future (rather than simpler), perhaps we'd want to introduce a message that traces each `ApiItem` as it is visited.  Or maybe we'd want a message that reports items that were visited but NOT added to the final data set (because they are occluded by a child class's override).  This is why I suggested a boolean to indicate actual data loss.

On the other hand, if you expect this crawl to become simpler in the future, such that the ***order in which things are crawled*** is not important information, then another idea would be to move this array of messages to `IFindMembersWithInheritanceResult` and eliminate the callback entirely.  Your intuition is probably more accurate than mine.